### PR TITLE
Improve search.in= boundaries

### PR DESCRIPTION
**Pull request by radare**
* Make it more flexible (support rwx & and ==)
* Fix search inside the debugger

### CODE REVIEW

**Comment by radare** [comment]
cc jacobrosenthal

**Comment by radare** [comment]
cc @jacobrosenthal

File: libr/core/cmd_search.c [hunk 1]
```diff
@@ -592,6 +592,21 @@ static void append_bound(RList *list, RIO *io, RInterval search_itv, ut64 from,
 	}
 }
 
+static bool maskMatches(int srwx, int mask, bool only) {
+	if (mask) {
+		if (only) {
+			if ((srwx & 7) != mask) {
```
Comment (ret2libc) [inline libr/core/cmd_search.c]: Just ‘return swrx &7 != mask’

File: libr/core/cmd_search.c [hunk 2]
```diff
@@ -592,6 +592,21 @@ static void append_bound(RList *list, RIO *io, RInterval search_itv, ut64 from,
...
+				return true;
+			}
+		} else {
+			if (!(srwx & mask)) {
```
Comment (ret2libc) [inline libr/core/cmd_search.c]: Same here

**Comment by jacobrosenthal** [comment]
This successfully duplicates my (now closed) pr +1

**Pull request merged**